pkg/apis/okctl.io/v1alpha1/cluster_v1alpha1.go

```diff
@@ -303,15 +306,50 @@ type ClusterDatabasesPostgres struct {
 
 // Validate the content of a postgres database declaration
 func (c ClusterDatabasesPostgres) Validate() error {
+	const minLengthDatabaseName int = 1
+
+	const maxLengthDatabaseName int = 60
+
+	/*
+	 * Validation of database name is based on:
+	 * 1. AWS RDS rules when creating a postgres database
+	 * 2. Limitations when we use cloudformation to create a bucket,
+	 *	capital letters will result in a error. AWS RDS will also lowercase all
+	 *	database names if you enter them with capital letters
+	 */
 	return validation.ValidateStruct(&c,
-		validation.Field(&c.Name, validation.Required, is.Alphanumeric),
-		validation.Field(&c.User, validation.Required, validation.NotIn([]string{
+		validation.Field(&c.Name,
+			validation.Required,
+			validation.NotIn(
+				"db",
+				"database",
+			).Error("'db' and 'database' are reserved postgres database names"),
+			validation.Length(minLengthDatabaseName, maxLengthDatabaseName).Error("database name cannot be longer than 60 characters"),
+			validation.Match(regexp.MustCompile("^[^A-Z]+$")).Error("database name cannot have capital letter"),
+			validation.Match(regexp.MustCompile("^[a-z]")).Error("database name must start with a letter"),
+			validation.Match(regexp.MustCompile("[^-]$")).Error("database name must not end with a hyphen"),
+			validation.By(fieldCanNotContainString("--", "database name can not have two consecutive hyphens")),
+		),
+		validation.Field(&c.User, validation.Required, validation.NotIn(
 			"admin",
-		})),
+		).Error("'admin' is a reserved postgres username")),
 		validation.Field(&c.Namespace, validation.Required),
 	)
 }
 
+func fieldCanNotContainString(str string, errorString string) validation.RuleFunc {
```

Comment:
I think the file readability could benifit from moving this function to it's own file in the current directory. Maybe `custom-validators.go`